### PR TITLE
feat: add Preview + Current Position to remove liquidity + some refacts

### LINKED ISSUE #177: Add remove liquidity preview info
When a user exits their liquidity, they need to be able to see what they currently have, and what they will get after they have exited.

### LINKED ISSUE #217: Add current position table for `Remove Liquidity` page
Add current position information when the user is removing liquidity.

Design link: https://miro.com/app/board/uXjVOynDXW4=/

### PR DESCRIPTION
**Pull request by LuizAsFight**
- Added `Preview Info` for `Remove Liquidity` page
- Added `Current Position` info for `Remove Liquidity` page
- Refact `Add Liquidity` preview to keep same pattern
- Fix `Current Position` component to show correctly pooled ETH / pooled DAI
- refact the way user data is organized in hooks (`useUserPosition`, `userPreviewAddLiquidity`, `userPreviewRemoveLiquidity`)
- improve math/bigint/number divide functions
- rename `PoolsPreview` page to `Pools`, as it's not a Preview

Closes #177 
Closes #217

### CODE REVIEW

File: packages/app/src/pages/PoolPage/PoolCurrentPosition.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { PreviewTable, PreviewItem } from "~/components/PreviewTable";
+import { TokenIcon } from "~/components/TokenIcon";
+import { useCoinMetadata } from "~/hooks/useCoinMetadata";
+import { useUserPositions } from "~/hooks/useUserPositions";
+
+export const PoolCurrentPosition = () => {
+  const info = useUserPositions();
+  const { coinMetaData } = useCoinMetadata({ symbol: "ETH/DAI" });
```
Comment (luizstacio) [inline packages/app/src/pages/PoolPage/PoolCurrentPosition.tsx:8]: We have a const for this

File: packages/app/src/pages/SwapPage/PricePerToken.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ function getPricePerToken(
   toAmount?: bigint | null
 ) {
   if (!toAmount || !fromAmount) return "";
-  const ratio = divideBigInt(toAmount, fromAmount);
+  const ratio = divideFnValidOnly(toAmount, fromAmount);
```
Comment (luizstacio) [inline packages/app/src/pages/SwapPage/PricePerToken.tsx:23]: What is valid only?

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/pages/SwapPage/PricePerToken.tsx (hunk 2 above)
Comment (LuizAsFight) [reply packages/app/src/pages/SwapPage/PricePerToken.tsx:23]: - `divideFn` do the raw division with the params you informed, which may result in invalid values, like `NaN` `Infinity` or even `undefined` 

- `divideFnValidOnly` handles inputs first, then do the raw division, then handle the output. Making sure that result is always gonna be a valid `number`

**Comment by luizstacio** [comment]
Can you do this also?
- When the user is viewing current positions and clicks on `remove liquidity` on the `CoinSelect`, it should show the name and the icon. The selector should be disabled also.
- The same thing should happen on `Add liqudity`

https://user-images.githubusercontent.com/3941923/170431438-6f7790b9-9768-48a3-91e8-8b89bed4e15f.mov

File: packages/app/src/pages/PoolPage/PoolCurrentPosition.tsx (hunk 1 above)
Comment (LuizAsFight) [reply packages/app/src/pages/PoolPage/PoolCurrentPosition.tsx:8]: could you tell where is it? I searched but only see other places using raw `ETH/DAI` but not using any default const param

Comment (luizstacio) [reply packages/app/src/pages/PoolPage/PoolCurrentPosition.tsx:8]: on file CoinsMeta

**Review by luizstacio — APPROVED** [review]

**Pull request merged**